opentelemetry-api/tests/util/test__importlib_metadata.py

```diff
@@ -21,6 +22,37 @@
 )
 
 
+class TestDependencies(TestCase):
+
+    def test_dependencies(self):
+
+        # pylint: disable=import-outside-toplevel
+        # pylint: disable=unused-import
+        # pylint: disable=import-error
+        if version_info.minor < 10:
+            try:
+                import importlib_metadata  # type: ignore
+
+            except ImportError:
+                self.fail(
+                    "importlib_metadata not installed when testing with "
+                    f"{version_info.major}.{version_info.minor}"
+                )
+
+        else:
+            try:
+                import importlib_metadata  # noqa
```

Comment:
Should this one and the fail message below be for `importlib.metadata` when Python 3.10+?

Reply:
Hmno, `importlib.metadata` comes with Python >= 3.10 natively:

```
(310) tigre@hilleman:~/sandbox/test_importlib_metadata$ pip freeze
Deprecated==1.2.14
-e git+ssh://git@github.com/ocelotl/opentelemetry-python.git@71dc0d2a33c7656e53e5cac0ebf1af95f3b5b9f9#egg=opentelemetry_api&subdirectory=opentelemetry-api
wrapt==1.16.0
(310) tigre@hilleman:~/sandbox/test_importlib_metadata$ python3
Python 3.10.14 (main, Sep  3 2024, 16:07:13) [GCC 11.4.0] on linux
Type "help", "copyright", "credits" or "license" for more information.
>>> import importlib.metadata
>>> 
```

What we are trying to test here is that when running Python < 3.10 `importlib_metadata` is installed and when running Python >= 3.10 `importlib_metadata` is not installed.

Reply:
Ohh got it. Thanks!